pkg/core/object/fmt.go

```diff
@@ -134,12 +134,7 @@ func (v *FormatValidator) Validate(obj *object.Object, unprepared bool) error {
 		} else {
 			// V2 split
 
-			if !firstSet {
-				// first part only
-				if obj.Parent() == nil {
-					return errors.New("v2 split: first object part does not have parent header")
-				}
-			} else {
+			if firstSet {
```

Comment:
why?

Reply:
I've expected this one. IIUC if `obj.HasParent()` now it can never have `obj.Parent() == nil` and semantically this makes total sense. Which in turn means that the whole `if !firstSet` branch turns no-op. See the respective test, it doesn't work with or without this code.

Reply:
forgot we are in `obj.HasParent()` branch